drivers/sensor/ina23x/ina237.c

```diff
@@ -63,7 +64,7 @@ static int ina237_channel_get(const struct device *dev, enum sensor_channel chan
 
 	case SENSOR_CHAN_POWER:
 		/* see datasheet "Current and Power calculations" section */
-		power_uw = (data->power * INA237_POWER_SCALING * config->current_lsb) / 10000U;
+		power_uw = (data->power * INA237_POWER_SCALING * config->current_lsb) / 10U;
```

Comment:
> There is no reason to make power_uw 64 bit here. It won't overflow until the power reaches 4.29 kW, which is far more than this sensor is designed to measure.

There is no fixed limit that the sensor measures since the INA237 uses an external shunt resistor, so the measurement limit is just based upon the data types involved.  Power is 24 bits, INA237_POWER_SCALING is 1 bit, and current_lsb can be up to 16 bits for a total of 41 bits in the numerator.

> It can be improved by changing the … * 2 / 10 into … / 5.

Good point, will change INA237_POWER_SCALING to 5, use it as a divisor, and will adjust the name.

Reply:
Fixed, will be in next patchset.

Reply:
@xyzzy42 PTAL